cypress/integration/page.spec.js

```diff
@@ -1,7 +1,193 @@
+    cy.get('@todos')
+      .eq(0)
+      .find('[type="checkbox"]')
+      .check();
+
+    cy.get('@todos')
+      .eq(0)
+      .should('have.class', 'completed');
+  });
+
+  it('Add ability to toggle the completed status of all the todos', () => {
+    cy.getByDataCy('toggleAll')
+      .check();
+
+    cy.get('@todos')
+      .children()
+      .each(() => {
+        cy.get('@todos')
+          .should('have.class', 'completed')
+      });
+
+    cy.getByDataCy('toggleAll')
+      .uncheck();
+
+    cy.get('@todos')
+      .children()
+      .each(() => {
+        cy.get('@todos')
+          .should('not.have.class', 'completed')
+      });
+  });
+
+  it('user can switch between `all`/`active`/`completed` todos', () => {
+    cy.get('@todos')
+      .eq(0)
+      .find('[type="checkbox"]')
+      .check();
+
+    cy.get('button')
+      .contains('Active')
+      .click();
+
+    cy.get('@todos')
+      .should('have.length', 2);
+
+    cy.get('button')
+      .contains('Completed')
+      .click();
+
+    cy.get('@todos')
+      .should('have.length', 1);
+
+    cy.get('button')
+      .contains('All')
+      .click();
+
+    cy.get('@todos')
+      .should('have.length', 3);
+  });
+
+  it('user can delete todos', () => {
+    cy.get('@todos')
+      .eq(0)
+      .find('[data-cy="deleteTodo"]')
+      .click({ force: true });
+
+    cy.get('@todos')
+      .should('have.length', 2);
+  });
+
+  it('user can clear the completed todos', () => {
+    cy.contains('Clear completed')
+      .should('not.exist');
+
+    cy.get('@todos')
+      .eq(0)
+      .find('[type="checkbox"]')
+      .check();
+
+    cy.get('button')
+      .contains('Clear completed')
+      .click();
+
+    cy.get('@todos')
+      .should('have.length', 2);
+  });
+
+  it('only input should be visible on the page when there are no todos', () => {
+    cy.getByDataCy('todosList').children().each(() => {
+      cy.get('@todos')
+        .find('[type="checkbox"]')
+        .check({ multiple: true })
+    });
+
+    cy.get('button')
+      .contains('Active')
+      .click();
+
+    cy.getByDataCy('todosList')
+      .should('exist');
+
+    cy.getByDataCy('todosFilter')
+      .should('be.visible');
+
+    cy.get('button')
+      .contains('All')
+      .click();
+
+    cy.get('button')
+      .contains('Clear completed')
+      .click();
+
+    cy.getByDataCy('todosList')
+      .should('not.exist');
+
+    cy.getByDataCy('todosFilter')
+      .should('not.exist');
+  });
+
+  it('user can edit todos', () => {
+    cy.get('@todos')
+      .eq(0)
+      .dblclick()
+      .type('!{enter}');
+
+    cy.get('@todos')
+      .eq(0)
+      .should('contain', 'buy some cheese!');
+
+    cy.get('@todos')
+      .eq(1)
+      .dblclick()
+      .type('!{esc}');
+
+    cy.get('@todos')
+      .eq(1)
+      .should('contain', 'feed the cat');
+
+    cy.get('@todos')
+      .eq(2)
+      .dblclick();
+
+    cy.get('@todos')
+      .eq(2).find('#editTodo').clear().type('{enter}');
```

Comment:
```suggestion
      .eq(2)
      .find('#editTodo')
      .clear()
      .type('{enter}');
```
також перевір ; всюди, де кінець команди

Reply:
ок